x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -134,52 +127,27 @@ func DelegatorSharesCallback(k Keeper, ctx sdk.Context, args []byte, query icqty
 //
 // We can prevent Case #1 by checking if the delegation total on the validator has changed
 //   while the query was in flight
-// We can prevent Case #2 by checking if there are any delegation unbonding records
-//   in state IN_PROGRESS (meaning an ICA is in flight)
+// We can prevent Case #2 by checking if the validator has a delegation change in progress
 func (k Keeper) CheckDelegationChangedDuringQuery(
```

Comment:
To sanity check the blocking loogic:

The check in this PR will ensure that there are no ICAs in flight when the ICQ comes back. 

It will NOT prevent a case like this:
- ICA begins, lands on host
- ICQ begins, hits the host
- ICA callback
- ICQ callback

But the above case should be prevented by the ICQ blocking the ICA, right?

Reply:
The case you described actually doesn't pose any risk because the acks landed on stride in the same order that they were landed on the host (i.e. ICA land, ICQ land, ICA ack, ICQ ack)

The main thing that matters in these permutations is the landing and ack'ing. The two concerning scenarios are shown [here](https://github.com/Stride-Labs/stride/blob/a8305beaf396a8541aebe041ddfd649f21b98382/x/stakeibc/keeper/icqcallbacks_delegator_shares.go#L120C1-L130)

Reply:
Ah right, thanks for clarifying